tls/ttls.c

```diff
@@ -224,12 +225,12 @@ ttls_skb_extract_alert(TlsIOCtx *io, TlsXfrm *xfrm)
 			continue;
 		}
 		n = min(skb->len - off, TTLS_ALERT_LEN - copied);
-		if (skb_copy_bits(skb, off, &io->alert[copied], n))
-			return T_BLOCK;
+		if ((r = skb_copy_bits(skb, off, &io->alert[copied], n)))
+			return r;
 		copied += n;
 	}
 
-	return skb ? 0 : T_BLOCK;
+	return skb ? 0 : T_BAD;
```

Comment:
Please attention may be you have more appropriate error code then T_BAD?

Reply:
Probably we're good with `T_BAD`, at least I don't see a better alternative as well as severe problems with this code